src/Network/Benchmark.hs

```diff
@@ -139,19 +138,20 @@ runAction
         , MonadBaseControl IO m
         , MonadIO m
         )
-    ⇒ Stat
+    ⇒ Int -- ^ number of retries
+    → Stat
     → TestAction
     → m Stat
-runAction stat action = timeT run >>= \case
-    (t, Right ()) →
-        return $ seq stat $ stat ⊕ successStat (toSeconds t * 1000)
-    (t, Left e) →
-        return $ seq stat $ stat ⊕ failStat (toSeconds t * 1000) (sshow e)
+runAction retries stat action = do
+    (t, result) ← timeT $ runEitherT run
+    return ∘ seq stat ∘ (stat ⊕) $
+        case result of
+            Right () → successStat $ toSeconds t * 1000
+            Left e → failStat (toSeconds t * 1000) (sshow e)
```

Comment:
no need to change, code readability at this level is probably mostly personal preference anyways.